app/app/controllers/webhooks/pinwheel/events_controller.rb

```diff
@@ -1,26 +1,46 @@
 class Webhooks::Pinwheel::EventsController < ApplicationController
-  DUMMY_API_KEY = "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"
-
+  before_action :authorize_webhook
   skip_before_action :verify_authenticity_token
 
-  def create
-    signature = request.headers["X-Pinwheel-Signature"]
-    timestamp = request.headers["X-Timestamp"]
+  DUMMY_API_KEY = "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"
+  EVENTS_MAP = {
+    "employment.added" => "employment_synced_at",
+    "income.added" => "income_synced_at",
+    "paystubs.fully_synced" => "paystubs_synced_at"
+  }
 
+  def authorize_webhook
     # To prevent timing attacks, we attempt to verify the webhook signature
     # using a same-length DUMMY_API_KEY even if the `end_user_id` does not
     # match a valid `cbv_flow`.
     cbv_flow = CbvFlow.find_by_pinwheel_end_user_id(params["payload"]["end_user_id"])
+    signature = request.headers["X-Pinwheel-Signature"]
+    timestamp = request.headers["X-Timestamp"]
+
     pinwheel = cbv_flow.present? ? pinwheel_for(cbv_flow) : PinwheelService.new(DUMMY_API_KEY, "sandbox")
     digest = pinwheel.generate_signature_digest(timestamp, request.raw_post)
     unless pinwheel.verify_signature(signature, digest)
-      return render json: { error: "Invalid signature" }, status: :unauthorized
+      render json: { error: "Invalid signature" }, status: :unauthorized
+    end
+  end
+
+  def create
+    cbv_flow = CbvFlow.find_by_pinwheel_end_user_id(params["payload"]["end_user_id"])
+
+    if cbv_flow && params["event"] == "account.added"
+      pinwheel_account = PinwheelAccount.find_or_create_by(pinwheel_account_id: params["payload"]["account_id"])
+      pinwheel_account.update!(cbv_flow: cbv_flow)
```

Comment:
Here's a fun possibility (optional):
```suggestion
      pinwheel_account = PinwheelAccount.create_with(cbv_flow: cbv_flow).find_or_create_by(pinwheel_account_id: params["payload"]["account_id"])
```

Reply:
Yeah that's right.